pkgs/tools/package-management/pdm/default.nix

```
@@ -24,41 +24,35 @@ in
 with python.pkgs;
 buildPythonApplication rec {
   pname = "pdm";
-  version = "1.14.0";
+  version = "2.1.1";
   format = "pyproject";
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-ZUbcuIRutSoHW5egCpwCKca2IZCgQsRAd72ueDzGySI=";
+    sha256 = "sha256-bffOjgRS6vh7TDD90GwTX2Iol5gWaFOf2+mEE5rUTWo=";
   };
 
-  # this patch allows us to run additional tests that invoke pdm, which checks
-  # itself for an update on every invocation by default, drammatically slowing
-  # down test runs inside the sandbox
-  #
-  # the patch is necessary because the fixture is creating a project and
-  # doesn't appear to respect the settings in `$HOME`; possibly a bug upstream
-  patches = [
-    ./check-update.patch
-  ];
-
   propagatedBuildInputs = [
     blinker
-    click
+    cachecontrol
     findpython
     installer
+    lockfile
     packaging
     pdm-pep517
     pep517
     pip
     platformdirs
     python-dotenv
-    pythonfinder
+    requests-toolbelt
     resolvelib
+    rich
     shellingham
     tomli
     tomlkit
+    unearth
+    virtualenv
   ] ++ lib.optionals (pythonOlder "3.8") [
```

```suggestion
  ]
  ++ cachecontrol.optional-dependencies.filecache
  ++ lib.optionals (pythonOlder "3.8") [
```

https://github.com/pdm-project/pdm/blob/main/pyproject.toml#L26